<!-- trajectory recorder launch -->
<launch >
	<!-- arguments -->
	<arg name="path_file" default="$(find trajectory_tools)/../../maps/nic_garage/sample_path.txt" />
	<arg name="use_simple_obstacle_avoidance" default="true" />

	<!-- trajectory follower -->
	<node name="trajectory_follower" pkg="trajectory_tools" type="trajectory_follower" output="screen" >
		<remap from="/twist_cmd" to="/twist_cmd_raw" if="$(arg use_simple_obstacle_avoidance)" />
		<param name="/map_frame" value="/world" type="string" />
		<param name="/base_link_frame" value="/base_link" type="string" />
		<param name="/path_file_name" value="$(arg path_file)" type="string" />
		<param name="/look_ahead_dist" value="2.5" type="double" />
		<param name="/max_vel" value="1.0" type="double" />
		<param name="/kv" value="0.7" type="double" />
		<param name="/publish_hz" value="20.0" type="double" />
	</node >

	<!-- obstacle avoidance -->
	<node name="simple_obstacle_avoidance" pkg="trajectory_tools" type="simple_obstacle_avoidance" output="screen" if="$(arg use_simple_obstacle_avoidance)" >
		<param name="/input_twist_topic_name" value="/twist_cmd_raw" type="string" />
		<param name="/input_scan_topic_name" value="/scan" type="string" />
		<param name="/output_twist_topic_name" value="/twist_cmd" type="string" />
		<param name="/local_map_size_x" value="10.0" type="double" />
		<param name="/local_map_size_y" value="10.0" type="double" />
		<param name="/local_map_pixel_size" value="0.05" type="double" />
		<param name="/robot_width" value="1.0" type="double" />
		<param name="/predict_time" value="5.0" type="double" />
		<param name="/step_time" value="0.1" type="double" />
		<param name="/kv" value="0.8" type="double" />
		<param name="/w_range" value="0.4" type="double" />
		<param name="/use_emergency_stop" value="true" type="bool" />
		<param name="/stop_x_size" value="0.6" type="double" />
		<param name="/stop_y_size" value="0.4" type="double" />
		<param name="/stop_time" value="3.0" type="double" />
		<param name="/back_time" value="5.0" type="double" />
		<param name="/back_speed" value="0.2" type="double" />
		<param name="/cmd_publish_hz" value="20.0" type="double" />
	</node >

	<!-- trajectory drawer -->
	<node name="trajectory_drawer" pkg="trajectory_tools" type="trajectory_drawer" >
		<param name="/map_frame" value="/world" type="string" />
		<param name="/path_file_name" value="$(arg path_file)" type="string" />
		<param name="/draw_interval" value="0.5" type="double" />
	</node >
</launch >
